examples/linear_model/plot_lasso_model_selection.py

```diff
@@ -1,171 +1,256 @@
+start_time = time.time()
+model = make_pipeline(StandardScaler(), LassoLarsCV(cv=20, normalize=False)).fit(X, y)
+fit_time = time.time() - start_time
 
-# Display results
-plt.figure()
-plt.semilogx(model.cv_alphas_ + EPSILON, model.mse_path_, ":")
+# %%
+plt.semilogx(model[-1].cv_alphas_, model[-1].mse_path_, ":")
 plt.semilogx(
-    model.cv_alphas_ + EPSILON,
-    model.mse_path_.mean(axis=-1),
-    "k",
+    model[-1].cv_alphas_,
+    model[-1].mse_path_.mean(axis=-1),
+    color="black",
     label="Average across the folds",
     linewidth=2,
 )
-plt.axvline(model.alpha_, linestyle="--", color="k", label="alpha CV")
-plt.legend()
+plt.axvline(model[-1].alpha_, linestyle="--", color="black", label="alpha CV")
 
+plt.ylim(ymin, ymax)
 plt.xlabel(r"$\alpha$")
 plt.ylabel("Mean square error")
-plt.title("Mean square error on each fold: Lars (train time: %.2fs)" % t_lasso_lars_cv)
-plt.axis("tight")
-plt.ylim(ymin, ymax)
-
-plt.show()
+plt.legend()
+_ = plt.title(f"Mean square error on each fold: Lars (train time: {fit_time:.2f}s)")
+
+# %%
+# Summary of cross-validation approach
+# ....................................
+# Both algorithms give roughly the same results.
+#
+# Lars computes a solution path only for each kink in the path. As a result, it
+# is very efficient when there are only of few kinks, which is the case if
+# there are few features or samples. Also, it is able to compute the full path
+# without setting any hyperparameter. On the opposite, coordinate descent
+# computes the path points on a pre-specified grid (here we use the default).
+# Thus it is more efficient if the number of grid points is smaller than the
+# number of kinks in the path. Such a strategy can be interesting if the number
+# of features is really large and there are enough samples to be selected in
+# each of the cross-validation fold. In terms of numerical errors, for heavily
+# correlated variables, Lars will accumulate more errors, while the coordinate
+# descent algorithm will only sample the path on a grid.
+#
+# Note how the optimal value of alpha varies for each fold. This illustrates
+# why nested-cross validation is necessary when trying to evaluate the
+# performance of a method for which a parameter is chosen by cross-validation:
+# this choice of parameter may not be optimal for a final evaluation on
+# unseen test set only.
+#
+# Conclusion
+# ----------
+# In this tutorial, we presented two approaches for selecting the best
+# hyperparameter `alpha`: one strategy finds the optimal value of `alpha`
+# by only using the training set and some information criterion, and another
+# strategy is based on cross-validation.
+#
+# In this example, both approaches are working similarly. The in-sample
+# hyperparameter selection even shows its efficacy in terms of computational
+# performance. However, it should be noted that it can only be used when the
+# number of samples is large enough compared to the number of features.
```

Comment:
Is this direct phrasing clearer?
```suggestion
# performance. However, it can only be used when the number of samples is large
# enough compared to the number of features.
```